pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/text/LuceneTextIndexReader.java

```diff
@@ -133,7 +131,8 @@ public MutableRoaringBitmap getDocIds(Object value) {
     MutableRoaringBitmap docIds = new MutableRoaringBitmap();
     Collector docIDCollector = new LuceneDocIdCollector(docIds, _docIdTranslator);
     try {
-      Query query = _queryParser.parse(searchQuery);
+      QueryParser parser = new QueryParser(_column, new StandardAnalyzer());
```

Comment:
Is there any way to reuse some components here? Will this cause too much garbage?
Please also add some comments so that future developers know that it cannot be reused.

Reply:
So the parser is JavaCC based and so it is stateful. We have to instantiate it per query (this is the bug). Analyzer on the other hand seems stateless so I reverted that part to be instantiated just once.